garminbackup.py

```diff
@@ -1,124 +1,32 @@
-              "./activities/"), default=os.path.join(".", "activities"))
-    parser.add_argument(
-        "--log-level", metavar="LEVEL", type=str,
-        help=("Desired log output level (DEBUG, INFO, WARNING, ERROR). "
-              "Default: INFO."), default="INFO")
-    parser.add_argument(
-        "-f", "--format", choices=export_formats,
-        default=None, action='append',
-        help=("Desired output formats ("+', '.join(export_formats)+"). "
-              "Default: ALL."))
-    parser.add_argument(
-        "-E", "--ignore-errors", action='store_true',
-        help="Ignore errors and keep going. Default: FALSE")
-    parser.add_argument(
-        "--max-retries", metavar="NUM", default=DEFAULT_MAX_RETRIES,
-        type=int, help="The maximum number of retries to make on failed attempts to fetch an activity. Exponential backoff will be used, meaning that the delay between successive attempts will double with every retry, starting at one second. DEFAULT: %d" % DEFAULT_MAX_RETRIES)
-
-    args = parser.parse_args()
-    if not args.log_level in LOG_LEVELS:
-        raise ValueError("Illegal log-level: {}".format(args.log_level))
-
-    # if no --format was specified, all formats are to be backed up
-    args.format = args.format if args.format else export_formats
-    log.info("backing up formats: %s", ", ".join(args.format))
-
+    args = parse_args()
     logging.root.setLevel(LOG_LEVELS[args.log_level])
 
     try:
-        if not os.path.isdir(args.backup_dir):
-            os.makedirs(args.backup_dir)
-
-        if not args.password:
-            args.password = getpass.getpass("Enter password: ")
-
-        # set up a retryer that will handle retries of failed activity
-        # downloads
-        retryer = Retryer(
-            delay_strategy=ExponentialBackoffDelayStrategy(
-                initial_delay=timedelta(seconds=1)),
-            stop_strategy=MaxRetriesStopStrategy(args.max_retries))
-
-
-        with GarminClient(args.username, args.password) as client:
-            # get all activity ids and timestamps from Garmin account
-            log.info("scanning activities for %s ...", args.username)
-            activities = set(retryer.call(client.list_activities))
-            log.info("account has a total of %d activities", len(activities))
-
-            missing_activities = garminexport.backup.need_backup(
-                activities, args.backup_dir, args.format)
-            backed_up = activities - missing_activities
-            log.info("%s contains %d backed up activities",
-                args.backup_dir, len(backed_up))
-
-            log.info("activities that aren't backed up: %d",
-                     len(missing_activities))
+        incremental_backup(username=args.username,
+                           password=args.password,
+                           backup_dir=args.backup_dir,
+                           format=args.format,
+                           log_level=args.log_level,
```

Comment:
I would say we skip the `log_level` argument here. Instead `incremental_backup` will simply use a logger named after its module (`garminexport.incremental_backup`) and it is up to the client to modify the log level for that logger if need be:
```
   logging.getLogger('garminexport.incremental_backup').setLevel(logging.ERROR)
```
or similar.

Reply:
`log_level` argument is gone. I did not notice that it is not used after refactoring